stdlib/InteractiveUtils/src/editless.jl

```diff
@@ -217,18 +220,29 @@ function editor()
 end
 
 """
-    edit(path::AbstractString, line::Integer=0, column::Integer=0)
+    edit(path::AbstractString, line::Integer=0, column::Integer=0; wait::Union{Bool, Nothing}=nothing)
 
 Edit a file or directory optionally providing a line number to edit the file at.
-Return to the `julia` prompt when you quit the editor. The editor can be changed
+
+`wait` can be `true`, `false` or `nothing`.
+
+- `wait = true`: return to the `julia` prompt when you quit the editor,
+- `wait = false`: return immediately while opening editor in a background task,
+- `wait = nothing`: default to whether to wait or not based on predefined editor callbacks.
+
+!!! warning
+    The values `true` or `false` for `wait` is not supported by all editors and
+    must be used on a case by case basis.
```

Comment:
Can you name a single editor for which they are "supported" then?

My understanding is that for editors that open in the terminal (e.g. plain `vi`) `wait=true` is required, and omitting it just won't allow editing.

While for GUI editors like `bbedit` or `code`, the command line tool often just notifies a separate GUI process to open a file and then exits immediately. For `code` could be changed by adding the `--wait` argument, but your PR doesn't do that and in fact as far as I can tell, can't do that in its current form: there is no way to use `define_editor` in a way that allows specifying "my editor supports both modes, you can choose.


I guess on Linux some GUI editors like `gedit` work by launching a new GUI process, so *there* wait/nowait perhaps has an effect?

But still: I would expect a PR like this one to not just randomly expose the `wait` flag to the user and say "hey, using this may or may not work, it is up to you to find out". Rather I'd expect there to be a genuine way to declare that an editor supports both modes? E.g. perhaps the `wait` argument for `define_editor` could become trivalent: `true`, `false`, or `:flexible`. And in that case, perhaps the invocation of `fn(cmd, path, line)` would become `fn(cmd, path, line, wait)`. Then one could e.g. change

```
    define_editor(["code", "code-insiders"]) do cmd, path, line, column
        `$cmd -g $path:$line:$column`
    end
```
to something like this:
```
    define_editor(["code", "code-insiders"]) do cmd, path, line, column, wait
        wait && return `$cmd --wait -g $path:$line:$column`
        `$cmd -g $path:$line:$column`
    end
```
or so

Reply:
My understanding was that a user can modify EDITOR_CALLBACKS to add a new editor for whatever they need it do, because we'd want the same editor to behave differently in different contexts (`Meta+e` versus `@edit some_fn()`).

For example, with this PR a user can change the `Meta+e` to push to EDITOR_CALLBACKS, call the default edit, and pop from EDITOR_CALLBACKS to restore the default behavior.

Reply:
Also, with neovim, it's possible to open a terminal in neovim itself. And in that case, it may make sense to use `wait` (e.g. https://github.com/mhinz/neovim-remote allows workflows where someone may want to set `nvr -cc split --remote-wait` as their EDITOR conditionally).

Reply:
Hmm, I think you are right. I'm not sure what exactly I tried when I made this PR, I thought I needed the `wait=true` exposed for this to work. I use neovim predominantly and was experimenting with vscode when I made this PR. 

I'm happy to convert this to a draft PR for now and revisit it again when I have time to look into this. I expect we'll revisit this feature anyway when the REPL 2.0 efforts move along.